app/models/follower.rb

```diff
@@ -0,0 +1,7 @@
+class Follower < ApplicationRecord
+  belongs_to :follower, foreign_key: 'follower_id', class_name: 'User'
+  belongs_to :following, foreign_key: 'following_id', class_name: 'User'
+
+  validates :follower, uniqueness: true
+  validates :following, uniqueness: true
```

Comment:
The validation should have been socped to validate the pair (follower, followee) is unique

```suggestion
  validates :follower, uniqueness: {scope: :following}
```